seb_openedx/middleware.py

```diff
@@ -38,16 +43,45 @@ def process_view(self, request, view_func, view_args, view_kwargs):
                 # Second: Granular white-listing
                 access_denied = True
 
+            masquerade, masquerade_html, user = self.handle_masquerade(request, course_key)  # pylint: disable=unused-variable
             active_comps = get_enabled_permission_classes()
             for permission in active_comps:
+                if masquerade and masquerade.role != 'staff' and permission == AlwaysAllowStaff:
+                    continue
                 if permission().check(request, course_key):
                     access_denied = False
 
-        if access_denied:
-            return self.handle_access_denied(request, view_func, view_args, view_kwargs, course_key)
+            if access_denied:
+                html = six.text_type(self.handle_access_denied(request, view_func, view_args, view_kwargs, course_key))
+                if masquerade:
+                    html += masquerade_html
+                http_response = HttpResponseForbidden(html)
+                return http_response
 
         return None
 
+    def supports_preview_menu(self, request):
+        """ if request view support preview_menu or not (not fully accurate) """
```

Comment:
I would not say this is "not fully accurate", it is more like: seb-openedx only supports masquerade on this and that views.
```suggestion
        """ Calculates if request view supports the preview_menu or not in this context. Not a perfect match to openedx's internal support"""
```